### ISSUE TITLE
RuntimeError / EOFError on import

### ISSUE THREAD

**Issue by ehultee** [bryanvriel/iceutils#10]
I am trying to use my dev branch, processing locally (Mac OSX), to extract a timeseries.  There seems to be a problem with multiprocessing setup in the environment.  I don't know what this has to do with the actual iceutils package, but it raises an error from the very first import.

```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/opt/anaconda3/envs/iu-gld/lib/python3.8/multiprocessing/spawn.py", line 116, in spawn_main
    exitcode = _main(fd, parent_sentinel)
  File "/opt/anaconda3/envs/iu-gld/lib/python3.8/multiprocessing/spawn.py", line 125, in _main
    prepare(preparation_data)
  File "/opt/anaconda3/envs/iu-gld/lib/python3.8/multiprocessing/spawn.py", line 236, in prepare
    _fixup_main_from_path(data['init_main_from_path'])
  File "/opt/anaconda3/envs/iu-gld/lib/python3.8/multiprocessing/spawn.py", line 287, in _fixup_main_from_path
    main_content = runpy.run_path(main_path,
  File "/opt/anaconda3/envs/iu-gld/lib/python3.8/runpy.py", line 263, in run_path
    return _run_module_code(code, init_globals, run_name,
  File "/opt/anaconda3/envs/iu-gld/lib/python3.8/runpy.py", line 96, in _run_module_code
    _run_code(code, mod_globals, init_globals,
  File "/opt/anaconda3/envs/iu-gld/lib/python3.8/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/Users/lizz/Documents/Research/Gld-timeseries/helheim-extract_ts.py", line 5, in <module>
    import iceutils as ice
  File "/opt/anaconda3/envs/iu-gld/lib/python3.8/site-packages/iceutils/__init__.py", line 12, in <module>
    from . import tseries
  File "/opt/anaconda3/envs/iu-gld/lib/python3.8/site-packages/iceutils/tseries/__init__.py", line 8, in <module>
    from .solver import *
  File "/opt/anaconda3/envs/iu-gld/lib/python3.8/site-packages/iceutils/tseries/solver.py", line 7, in <module>
    import pymp
  File "/opt/anaconda3/envs/iu-gld/lib/python3.8/site-packages/pymp/__init__.py", line 14, in <module>
    import pymp.shared as _shared
  File "/opt/anaconda3/envs/iu-gld/lib/python3.8/site-packages/pymp/shared.py", line 15, in <module>
    _MANAGER = _multiprocessing.Manager()  # pylint: disable=no-member
  File "/opt/anaconda3/envs/iu-gld/lib/python3.8/multiprocessing/context.py", line 57, in Manager
    m.start()
  File "/opt/anaconda3/envs/iu-gld/lib/python3.8/multiprocessing/managers.py", line 579, in start
    self._process.start()
  File "/opt/anaconda3/envs/iu-gld/lib/python3.8/multiprocessing/process.py", line 121, in start
    self._popen = self._Popen(self)
  File "/opt/anaconda3/envs/iu-gld/lib/python3.8/multiprocessing/context.py", line 283, in _Popen
    return Popen(process_obj)
  File "/opt/anaconda3/envs/iu-gld/lib/python3.8/multiprocessing/popen_spawn_posix.py", line 32, in __init__
    super().__init__(process_obj)
  File "/opt/anaconda3/envs/iu-gld/lib/python3.8/multiprocessing/popen_fork.py", line 19, in __init__
    self._launch(process_obj)
  File "/opt/anaconda3/envs/iu-gld/lib/python3.8/multiprocessing/popen_spawn_posix.py", line 42, in _launch
    prep_data = spawn.get_preparation_data(process_obj._name)
  File "/opt/anaconda3/envs/iu-gld/lib/python3.8/multiprocessing/spawn.py", line 154, in get_preparation_data
    _check_not_importing_main()
  File "/opt/anaconda3/envs/iu-gld/lib/python3.8/multiprocessing/spawn.py", line 134, in _check_not_importing_main
    raise RuntimeError('''
RuntimeError: 
        An attempt has been made to start a new process before the
        current process has finished its bootstrapping phase.

        This probably means that you are not using fork to start your
        child processes and you have forgotten to use the proper idiom
        in the main module:

            if __name__ == '__main__':
                freeze_support()
                ...

        The "freeze_support()" line can be omitted if the program
        is not going to be frozen to produce an executable.
Traceback (most recent call last):
  File "helheim-extract_ts.py", line 5, in <module>
    import iceutils as ice
  File "/opt/anaconda3/envs/iu-gld/lib/python3.8/site-packages/iceutils/__init__.py", line 12, in <module>
    from . import tseries
  File "/opt/anaconda3/envs/iu-gld/lib/python3.8/site-packages/iceutils/tseries/__init__.py", line 8, in <module>
    from .solver import *
  File "/opt/anaconda3/envs/iu-gld/lib/python3.8/site-packages/iceutils/tseries/solver.py", line 7, in <module>
    import pymp
  File "/opt/anaconda3/envs/iu-gld/lib/python3.8/site-packages/pymp/__init__.py", line 14, in <module>
    import pymp.shared as _shared
  File "/opt/anaconda3/envs/iu-gld/lib/python3.8/site-packages/pymp/shared.py", line 15, in <module>
    _MANAGER = _multiprocessing.Manager()  # pylint: disable=no-member
  File "/opt/anaconda3/envs/iu-gld/lib/python3.8/multiprocessing/context.py", line 57, in Manager
    m.start()
  File "/opt/anaconda3/envs/iu-gld/lib/python3.8/multiprocessing/managers.py", line 583, in start
    self._address = reader.recv()
  File "/opt/anaconda3/envs/iu-gld/lib/python3.8/multiprocessing/connection.py", line 250, in recv
    buf = self._recv_bytes()
  File "/opt/anaconda3/envs/iu-gld/lib/python3.8/multiprocessing/connection.py", line 414, in _recv_bytes
    buf = self._recv(4)
  File "/opt/anaconda3/envs/iu-gld/lib/python3.8/multiprocessing/connection.py", line 383, in _recv
    raise EOFError
EOFError
```

Something you've seen before?

**Comment by bryanvriel** [comment]
That looks to be an issue with Python 3.8 and `pymp`. I recreated your environment on my laptop, and I get the same error with this minimum working example script:
```python
#!/usr/bin/env python3

import pymp
import sys

def main():
    print('hello world')

if __name__ == '__main__':
    main()
```
I looked into the `pymp` code, and I can see where the use of `multiprocessing` violates the current Python 3.8 idiom.

I went ahead and forked the `pymp` repository and made some changes to the code. Checkout this branch [here](https://github.com/bryanvriel/pymp/tree/py38). Install it manually using `python3 setup.py install`, and then see if the above test script works out for you.

**Comment by ehultee** [comment]
I installed your forked version of `pymp` into my local environment.  Ran the test script and it produces the same error.

When you say you recreated my environment, do you mean you copied a package list from somewhere on mrschippy?  Where I am getting the error is in my *local* environment.  I'm attaching a requirements list from pip freeze.

[iu_gld-requirements.txt](https://github.com/bryanvriel/iceutils/files/4530465/iu_gld-requirements.txt)

**Comment by bryanvriel** [comment]
Uh oh. Can you double check that the code in the `pymp/shared.py` file matches the changes I made in my `pymp` master branch? You can find the location in python using:
```
import pymp
print(pymp.shared.__file__)
```
When I originally posted this comment, I linked to the wrong branch, so hopefully that didn't throw you off.

For your environment, I installed a clean version of Miniconda on my laptop, created a virtual environment with python 3.8, and installed the prerequisites for iceutils. It looks like our pip freeze outputs are identical, except for tqdm.

**Comment by ehultee** [comment]
Okay, indeed the code didn't match.  I don't really understand why, since I had cloned your `pymp` repository to local and it doesn't show any changes.  Anyway, I removed and reinstalled it.  Now the code in `pymp/shared.py` matches and the minimal test script works.

I think this resolves the error for the moment.  It might be worth leaving the issue open, though, since for now the user still has to do a special package installation (not via conda or pip) in addition to the main `iceutils` setup.
